src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,1001 @@
+                }
+            }
+        }
+    },
+    dynamicComponents: {
+        settingsTables: {
+            createOnEvent: "createSettingsTable",
+            type: "gpii.devpmt.settingsTableWidget",
+            container: "{arguments}.0",
+            options: {
+                selectors: {
+                    initial: ".pmt-settings-table-div"
+                },
+                appId: "{arguments}.1",
+                solution: "{arguments}.2",
+                appUri: "{arguments}.3",
+                model: {
+                    flatPrefs: "{gpii.devpmt.editPrefs}.model.flatPrefs",
+                    contextNames: "{gpii.devpmt.editPrefs}.model.contextNames"
+                },
+                modelRelay: {
+                    settingsFilter: {
+                        source: {
+                            context: "gpii.devpmt.editPrefs",
+                            segs: ["productTableFilters", "{that}.options.appId", "settingsFilter"]
+                        },
+                        target: "settingsFilter",
+                        singleTransform: {
+                            type: "fluid.transforms.identity"
+                        }
+                    },
+                    allSettingsEnabled: {
+                        source: {
+                            context: "gpii.devpmt.editPrefs",
+                            segs: ["productTableFilters", "{that}.options.appId", "allSettingsEnabled"]
+                        },
+                        target: "allSettingsEnabled",
+                        singleTransform: {
+                            type: "fluid.transforms.identity"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    selectors: {
+        initial: "#pmt-editprefset-viewport",
+        modalDialogContainer: "#modal-dialog-container",
+        modalDialogRender: "#pmt-modal-dialog-render",
+        topNavBar: "#pmt-topNavBar-container",
+        editWidgetSidebar: "#pmt-editwidget-sidebar",
+        tableOfContentsContainer: "#pmt-editPrefsTOC-container",
+        productListContainer: "#pmt-productList-container",
+        prefsSafeInfoContainer: "#pmt-prefsSafeInfo-container",
+        prefsSafeKeysListContainer: "#pmt-prefsSafeKeysList-container",
+        prefSetsListContainer: "#pmt-prefSetsList-container",
+        prefsAdjusterContainer: "#pmt-prefs-adjuster-container",
+        genericSettingsTableContainer: "#pmt-genericSettingsTable-container",
+        // Each product table has this class
+        eachProductArea: ".pmt-single-product-area",
+        flatPrefsJsonDev: "#pmt-flatPrefs-jsonDev-container",
+        sidebarAccordian: "#pmt-sidebar-accordian",
+        valueDisplayCell: ".pmt-value-display",
+        foundationSticky: ".sticky"
+    },
+    templates: {
+        initial: "editprefset-viewport",
+        editPrefWidget: "editpref-widget"
+    },
+    invokers: {
+        prefsetExists: {
```

Comment:
`gpii.devpmt.editPrefs` is quite loaded. As this function is only used by `gpii.devpmt.dialogs.editContextDialog` component, it might be helpful to move this function to there.

Reply:
The comment above applies to other functions defined in `gpii.devpmt.editPrefs`.

Reply:
Actually, the body of this function was already in the utils code, so I've just removed this invoker, and calling it by funcName in the dialog without an invoker.